### PR TITLE
[#91] Add nested interpolation cop

### PR DESCRIPTION
**Pull request by roman-dubrovsky**
Resolves issue #91 . Also, I'm going to send this cop to the rubocop repo. 

Before you submit a pull request, please make sure you have to follow:

- [x] read and know items from the [Contributing Guide](https://github.com/datarockets/datarockets-style/blob/master/CONTRIBUTING.md#pull-requests)
- [x] add a description of the problem you're trying to solve (short summary from related issue)
- [x] verified that cops are ordered by alphabet
- [x] add a note to the style guide docs (if it needs)
- [x] add a note to the changelog file
- [x] the commit message contains the number of the related issue (if it presents)
  and word `Fix` if this PR closes related issue
- [ ] squash all commits before submitting to review

### CODE REVIEW

File: spec/datarockets/style/cop/style/nested_interpolation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  it "accepts two single interpolation" do
+    expect_no_offenses('"Hello #{user1.name} and #{user2.name}"')
+  end
+
+  it 'accepts interpolation with or logic' do
+    expect_no_offenses('"\n#{(lines.join("\n").split(node.source).first || \'\')}"')
+  end
+
+  it 'accepts interpolation into business block' do
+    expect_no_offenses('"#{assignment.join("\n#{indentation(node)}")}"')
```
Comment (roman-dubrovsky) [inline spec/datarockets/style/cop/style/nested_interpolation_spec.rb]: How do you think should we ban that construction? Or we can accept it?

Comment (texpert) [reply spec/datarockets/style/cop/style/nested_interpolation_spec.rb]: @roman-dubrovsky , I think it is redundant and should be banned.

**Comment by roman-dubrovsky** [comment]
@lazycoder9 heh, tests are red 😄

**Comment by lazycoder9** [comment]
> @lazycoder9 heh, tests are red 😄

Yeah, noted it after approve :(

**Pull request merged**